Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp

```diff
@@ -726,14 +810,22 @@ class Q_DECL_EXPORT Scene_aabb_plane_item : public Scene_plane_item
 
           // get sign through ray casting (random vector)
           Simple_kernel::Ray_3  ray(p, random_vec);
-          size_type nbi = min_tree->number_of_intersected_primitives(ray);
+          size_type nbi = (is_min_sm) ? min_sm_tree->number_of_intersected_primitives(ray)
+                                      : min_tree->number_of_intersected_primitives(ray);
 
           FT sign ( (nbi&1) == 0 ? 1 : -1);
           m_distance_function[i][j].second = sign * unsigned_distance;
         }
+      }
     }
 #else
-    FillGridSize<Tree> f(m_grid_size, diag, m_distance_function, m_max_distance_function, trees, is_signed, frame);
+    Q_FOREACH(Tree *tree, trees->values())
+    if(!(is_signed && !qobject_cast<Scene_polyhedron_item*>(trees->key(tree))->polyhedron()->is_closed()))
+      closed_trees.push_back(tree);
+    Q_FOREACH(SM_Tree *sm_tree, sm_trees->values())
+    if(!(is_signed && !CGAL::is_closed(*qobject_cast<Scene_surface_mesh_item*>(sm_trees->key(sm_tree))->polyhedron())))
+      closed_sm_trees.push_back(sm_tree);
+    FillGridSize<Tree, SM_Tree> f(m_grid_size, diag, m_distance_function, m_max_distance_function, closed_trees, closed_sm_trees, is_signed, frame);
```

Comment:
To reduce the duplication of code, the sequential version should use the functor `FillGridSize<Tree, SM_Tree>` as well.